code/scripting/api/libs/mission.cpp

```diff
@@ -955,6 +989,99 @@ ADE_FUNC(createWeapon,
 		return ade_set_error(L, "o", l_Weapon.Set(object_h()));
 }
 
+ADE_FUNC(createWarpeffect,
+	l_Mission,
+	"vector WorldPosition, vector PointTo, number radius, number duration /* Must be >= 4*/, fireballclass Class, "
+	"soundentry WarpOpenSound, soundentry WarpCloseSound, "
+	"[number WarpOpenDuration = -1, number WarpCloseDuration = -1, vector Velocity /* null vector by default*/, "
+	"boolean Use3DModel = false]",
+	"Creates a warp-effect fireball and returns a handle to it.",
+	"fireball",
+	"Fireball handle, or invalid fireball handle if fireball couldn't be created.")
+{
+	vec3d pos = vmd_zero_vector;
+	vec3d point_to = vmd_zero_vector;
+	float radius = 0.0f;
+	float duration = 4.0f;
+	int fireballclass = -1;
+	sound_entry_h *opensound = NULL;
+	sound_entry_h *closesound = NULL;
+
+	float opentime = -1.0f;
+	float closetime = -1.0f;
+	vec3d velocity = vmd_zero_vector;
+	bool warp3d = false;
+
+	if (!ade_get_args(L, "ooffooo|ffob", l_Vector.Get(&pos), l_Vector.Get(&point_to), &radius, &duration, l_Fireballclass.Get(&fireballclass), l_SoundEntry.GetPtr(&opensound), l_SoundEntry.GetPtr(&closesound), &opentime, &closetime, l_Vector.Get(&velocity), &warp3d)) {
+		return ade_set_error(L, "o", l_Fireball.Set(object_h()));
+	}
+
+	int flags = warp3d ? FBF_WARP_VIA_SEXP | FBF_WARP_3D : FBF_WARP_VIA_SEXP;
+
+	duration = duration < 4.0f ? 4.0f : duration;
+
+	// sanity check, if these were specified
+	if (duration < opentime + closetime)
+	{
+		//Both warp opening and warp closing must occur within the duration of the warp effect.
+		LuaError(L, "The duration of the warp effect must be higher than the sum of the opening and close durations");
+	}
```

Comment:
The duration check should also cause an error when that duration check fails. The same goes for the `IS_VEC_NULL_SQ_SAFE` check a bit below this.

Also, `LuaError` can still return normally if the user ignored the error so your previous error correction could would still be good to have here.

Reply:
Ah, thanks for the clarification